### ISSUE TITLE
Model trained well with 97.6% accuracy however the results are bad

### ISSUE THREAD

**Issue by gsareen86** [modnova/VAE-LSTM#2]
I'm able to successfully train the model with around 10,000 iterations on data (MSCoco + Quora). However, when I try to test the predictions on following sentences, the results are weird. Plz let me know if I'm testing the model correctly.

Statement used for testing the predictions:
python engine/train_rvae.py --mode=predict --model_dir=E:\Python\GitHub\VAE-LSTM-master\src\models\Gaurav_Test\export --data_path=E:\Python\GitHub\VAE-LSTM-master\src\models\Gaurav_Test\export\Testing_File.txt

Input: 
Text file containing following sentences:
> An angled view of a beautifully decorated bathroom.
> A clock that blends in with the wall hangs in a bathroom.
> A panoramic photo of a kitchen and dining room

Output: 
![image](https://user-images.githubusercontent.com/7555186/50517631-32eab600-0ad7-11e9-94d3-02581cdf60a1.png)

**Comment by petalrame** [comment]
I was never able to fully get the model to train correctly. During the training phase, it looks like it trains well, but if you take a look at the evaluation run and see the accuracy and loss in TensorBoard, you will see that the model overfits quite a lot and to be honest, I'm not quite sure why that is. I've taken a look at the execution graph and it all looks correct, except for the uninitialized warnings in TB(although I don't think this is the issue). I've also tried playing around with the loss function, but I don't think I've gotten it quite right. For reference on why the loss function is built that way you should take a look at the references(specifically [this](https://github.com/kefirski/pytorch_RVAE)). I didn't get a chance to fix it for a while, but I'll look into it again. In the meantime, if you find anything(fixes, more bugs, etc.) and let me know, I'd really appreciate it! 

**TL;DR**: I don't know what's wrong with it. Look at the loss in training and eval via TensorBoard, if it overfits then there is something wrong. I would suggest looking at the loss function(see rvae.py and references) and the execution graph. If you find out why it spits out gibberish lemme know, but I'll try to look into it in the meantime.
